src/Impostor.Server/Net/Inner/Objects/InnerMeetingHud.PlayerVoteArea.cs

```diff
@@ -34,34 +29,34 @@ public PlayerVoteArea(InnerMeetingHud parent, InnerPlayerInfo targetPlayer, bool
 
             public bool DidReport { get; private set; }
 
-            public sbyte VotedForId
+            public byte VotedForId
             {
                 get => _votedForId;
 
                 private set
                 {
                     _votedForId = value;
 
-                    if (DidVote)
-                    {
-                        switch ((VoteType)value)
-                        {
-                            case TVoteType.ForceSkip:
-                            case TVoteType.Skip:
-                                VoteType = (VoteType)value;
-                                break;
-
-                            default:
-                                VoteType = TVoteType.Player;
-                                VotedFor = Parent.Game.GameNet.GameData!.GetPlayerById((byte)value)?.Controller;
-                                break;
-                        }
-                    }
-                    else
+                    switch ((VoteType)value)
                     {
-                        VoteType = null;
-                        VotedFor = null;
+                        case TVoteType.Dead:
+                            IsDead = true;
+                            VoteType = TVoteType.Dead;
+                            break;
+
+                        case TVoteType.HasNotVoted:
+                        case TVoteType.Missed:
+                        case TVoteType.Skipped:
+                            VoteType = (VoteType)value;
+                            break;
+
+                        default:
+                            VoteType = TVoteType.Player;
+                            VotedFor = Parent.Game.GameNet.GameData!.GetPlayerById(value)?.Controller;
+                            break;
                     }
+
+                    DidVote = VoteType != TVoteType.HasNotVoted;
```

Comment:
why store it as a separate property instead of implementing a getter that does the compare then?

idem for IsDead